### PR TITLE
Remove useDynamicScale option

### PR DESCRIPTION
**Pull request by oterral**
# How to

This useDynamicScale function is application specific. There is no way that every user needs that.
If someone wants the same functionnality he can implements his own layer and override the functions.

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] The new class' members & methods are well documented
- [x] Tests added.

### CODE REVIEW

**Review by friedjoff — APPROVED** [review]
🎉 less project-specific code 🥳

File: src/api/tralis/TralisAPI.js [hunk 1]
```diff
@@ -102,17 +102,20 @@ class TralisAPI {
    *
    * @param {string} channel Name of the websocket channel to unsubscribe.
    * @param {string} [suffix=''] Suffix to add to the channel name.
+   * @param {function} cb The callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:105]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 2]
```diff
@@ -220,10 +223,11 @@ class TralisAPI {
 
   /**
    * Unsubscribe from current departures channel.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:226]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 3]
```diff
@@ -241,9 +245,10 @@ class TralisAPI {
 
   /**
    * Unsubscribe disruptions.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:248]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 4]
```diff
@@ -324,10 +329,11 @@ class TralisAPI {
 
   /**
    * Unsubscribe to stations channel.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:332]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 5]
```diff
@@ -357,9 +363,10 @@ class TralisAPI {
 
   /**
    * Unsubscribe to extra_geoms channel.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:366]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 6]
```diff
@@ -369,15 +376,16 @@ class TralisAPI {
...
    */
   subscribeTrajectory(mode, onMessage) {
-    this.unsubscribeTrajectory();
+    this.unsubscribeTrajectory(onMessage);
     this.subscribe(`trajectory${getModeSuffix(mode, TralisModes)}`, onMessage);
   }
 
   /**
    * Unsubscribe to trajectory channels.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:385]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 7]
```diff
@@ -396,9 +404,10 @@ class TralisAPI {
 
   /**
    * Unsubscribe to deleted_vhicles channels.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:407]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 8]
```diff
@@ -462,9 +471,10 @@ class TralisAPI {
    * Unsubscribe from full_trajectory channel
    *
    * @param {number} id A vehicle id.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:474]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/TralisAPI.js [hunk 9]
```diff
@@ -531,9 +541,10 @@ class TralisAPI {
    * Unsubscribe from stopsequence channel
    *
    * @param {number} id A vehicle id.
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/TralisAPI.js:544]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

File: src/api/tralis/WebSocketConnector.js [hunk 10]
```diff
@@ -176,18 +177,24 @@ class WebSocketConnector {
...
+      this.send(`GET ${reqStr}`);
+      this.send(`SUB ${reqStr}`);
+      this.subscribed[reqStr] = true;
+    }
   }
 
   /**
    * Unsubscribe from a channel.
    * @param {string} source source to unsubscribe from
+   * @param {function} cb The listener callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribe.
```
Comment (friedjoff) [inline src/api/tralis/WebSocketConnector.js:190]:
```suggestion
   * @param {function} cb Callback function to unsubscribe. If null all subscriptions for the channel will be unsubscribed.
```

**Review by friedjoff — APPROVED** [review]
Code LGTM.

Just some small suggestions to make documentation a bit more consistent.

**Comment by oterral** [comment]
Thx @friedjoff

**Pull request merged**